tensorrt_llm/_torch/pyexecutor/sampling_utils.py

```diff
@@ -266,7 +266,8 @@ def greedy_search_sampling_batch(
     next_tokens = torch.argmax(logits, dim=-1)
     softmax: Optional[torch.Tensor] = None
     if return_probs:
-        softmax = torch.softmax(logits, dim=-1)
+        softmax = torch.zeros_like(logits)
+        softmax.scatter_(1, next_tokens.unsqueeze(-1), 1.0)
```

Comment:
I think this changes the semantics, when applying softmax, the probs for other tokens not being select is the value of softmax function or just 0?

How does other FW do?